pkg/engine/eventsink.go

```diff
@@ -37,65 +37,96 @@ type eventSink struct {
 	statusSink bool         // whether this is an event sink for status messages.
 }
 
-func (s *eventSink) Logf(sev diag.Severity, d *diag.Diag, args ...interface{}) {
+func (s *eventSink) LogfDepth(
+	depth int,
+	sev diag.Severity,
+	d *diag.Diag,
+	args ...interface{}) {
+
 	switch sev {
 	case diag.Debug:
-		s.Debugf(d, args...)
+		s.DebugfDepth(depth+1, d, args...)
 	case diag.Info:
-		s.Infof(d, args...)
+		s.InfofDepth(depth+1, d, args...)
 	case diag.Infoerr:
-		s.Infoerrf(d, args...)
+		s.InfoerrfDepth(depth+1, d, args...)
 	case diag.Warning:
-		s.Warningf(d, args...)
+		s.WarningfDepth(depth+1, d, args...)
 	case diag.Error:
-		s.Errorf(d, args...)
+		s.ErrorfDepth(depth+1, d, args...)
 	default:
 		contract.Failf("Unrecognized severity: %v", sev)
 	}
 }
 
-func (s *eventSink) Debugf(d *diag.Diag, args ...interface{}) {
+func (s *eventSink) DebugfDepth(depth int, d *diag.Diag, args ...interface{}) {
```

Comment:
Just curious: do you expect that we may call these methods with `depth > 1` in the future?

Reply:
Eh, nothing thoughtful here, if it's unused I'd like to delete. Let me try deleting.